File: backend/gateway/routes/health.py
"""
Health check endpoints
Based on monitoring-service.md specification
"""

from fastapi import APIRouter, Depends
from fastapi.responses import JSONResponse
import structlog
from datetime import datetime

logger = structlog.get_logger(__name__)

router = APIRouter()


@router.get("/live")
async def liveness_check():
    """Kubernetes liveness probe endpoint"""
    return {"status": "alive", "timestamp": datetime.utcnow().isoformat()}


@router.get("/ready")
async def readiness_check():
    """Kubernetes readiness probe endpoint"""
    # TODO: Add actual health checks for dependencies
    # - Database connectivity
    # - Redis connectivity  
    # - Downstream service availability
    
    checks = {
        "database": "healthy",  # TODO: Implement actual check
        "redis": "healthy",     # TODO: Implement actual check
        "services": "healthy"   # TODO: Implement actual check
    }
    
    all_healthy = all(status == "healthy" for status in checks.values())
    
    return JSONResponse(
        status_code=200 if all_healthy else 503,
        content={
            "status": "ready" if all_healthy else "not_ready",
            "timestamp": datetime.utcnow().isoformat(),
            "checks": checks
        }
    )


@router.get("/")
async def health_status():
    """General health status endpoint"""
    return {
        "service": "OET Praxis API Gateway",
        "version": "1.0.0",
        "status": "healthy",
        "timestamp": datetime.utcnow().isoformat()
    }

File: backend/services/user-service/src/routes/health.ts
/**
 * Health check routes for User Service
 */

import { Router } from 'express';
import { asyncHandler } from '../middleware/error-handler';

const router = Router();

router.get('/live', asyncHandler(async (req, res) => {
  res.json({
    status: 'alive',
    service: 'user-service',
    version: '1.0.0',
    timestamp: new Date().toISOString(),
  });
}));

router.get('/ready', asyncHandler(async (req, res) => {
  // TODO: Add actual readiness checks
  // - Database connectivity
  // - Redis connectivity
  const checks = {
    database: 'healthy',
    redis: 'healthy',
  };

  const allHealthy = Object.values(checks).every(status => status === 'healthy');

  res.status(allHealthy ? 200 : 503).json({
    status: allHealthy ? 'ready' : 'not_ready',
    service: 'user-service',
    checks,
    timestamp: new Date().toISOString(),
  });
}));

export default router;